distributed/shuffle/_shuffle.py

```diff
@@ -288,10 +289,10 @@ def _construct_graph(self) -> _T_LowLevelGraph:
         name = "shuffle-transfer-" + token
         transfer_keys = list()
         for i in range(self.npartitions_input):
-            transfer_keys.append((name, i))
-            dsk[(name, i)] = (
+            t = Task(
+                (name, i),
                 shuffle_transfer,
-                (self.name_input, i),
+                TaskRef((self.name_input, i)),
```

Comment:
The `test_merge` tests w/ dask-expr enabled never take this code path. That's interesting but not incredibly surprising.